aptos-move/e2e-move-tests/src/tests/scripts.rs

```diff
@@ -146,6 +146,45 @@ fn test_script_with_type_parameter() {
     assert_success!(status);
 }
 
+#[test]
+fn test_script_with_signer_parameter() {
+    let mut h = MoveHarness::new();
+
+    let alice = h.new_account_at(AccountAddress::from_hex_literal("0xa11ce").unwrap());
+
+    let package = build_package(
+        common::test_dir_path("script_with_signer.data/pack"),
+        aptos_framework::BuildOptions::default(),
+    )
+    .expect("building package must succeed");
+
+    let code = package.extract_script_code().into_iter().next().unwrap();
+
+    let txn = TransactionBuilder::new(alice.clone())
+        .script(Script::new(code, vec![], vec![
+            TransactionArgument::U64(0),
+            TransactionArgument::Serialized(
+                MoveValue::Signer(*alice.address())
+                    .simple_serialize()
+                    .unwrap(),
+            ),
+        ]))
+        .sequence_number(10)
+        .max_gas_amount(1_000_000)
+        .gas_unit_price(1)
+        .sign();
+
+    let status = h.run(txn);
+    assert_eq!(
+        status,
+        TransactionStatus::Keep(
+            aptos_types::transaction::ExecutionStatus::MiscellaneousError(Some(
+                aptos_types::vm_status::StatusCode::INVALID_MAIN_FUNCTION_SIGNATURE
```

Comment:
Why are we adding this test? Has this not been tested before?

Separately: would passing int, signer, signer here be allowed (re-ordered from function signature)?

Reply:
It was just to make sure the signer gets rejected if it occurs in the middle of arguments, thus it's safe for us to change the serialization format.